libs/challenge-registry/challenge-search/src/lib/challenge-search.component.ts

```diff
@@ -281,4 +324,21 @@ export class ChallengeSearchComponent implements OnInit, AfterContentInit {
       );
     }
   }
+
+  // mock up the service to filter challenge by search term
+  private searchChallenges(
+    challenges: Challenge[],
+    searchTerm: string
+  ): Challenge[] {
+    return challenges.filter((challenge) =>
+      (Object.keys(challenge) as [keyof Challenge]).some((k) =>
```

Comment:
Not sure if this is something we should address here or in a separate ticket, but --

When I search with `dream` (no space), no filtering seems to be done and all 4 cards are still displayed:

![Screen Shot 2023-01-06 at 2 31 13 PM](https://user-images.githubusercontent.com/9377970/211112466-9277d22c-f34f-4cc1-867a-68567882b435.png)

After some sleuthing, I see that this is due to "dream" is part of the [`fullName` attribute](https://github.com/rrchai/sage-monorepo/blob/32ea7d5e057b36fafec2ef0c2c7658928aaf4e16/libs/challenge-registry/ui/src/lib/challenge-card/mock-challenges.ts#L19) for the 4 mock challenges.

If I instead search with `dream ` (with a space) or use the DREAM facet, I get only one card back:

![Screen Shot 2023-01-06 at 2 31 19 PM](https://user-images.githubusercontent.com/9377970/211112673-20080e1b-30a6-4541-ad1c-5eddc3b5e834.png)

Thinking as a user, getting the one card back will make the most sense to me.

---

Basically, my question is:
Are we still using the `fullName` attribute?  I don't see it listed anymore in [challenge.openapi.yaml](https://github.com/Sage-Bionetworks/sage-monorepo/blob/main/libs/challenge-registry/api-description/build/challenge.openapi.yaml).

Reply:
> After some sleuthing, I see that this is due to "dream" is part of the [fullName attribute](https://github.com/rrchai/sage-monorepo/blob/32ea7d5e057b36fafec2ef0c2c7658928aaf4e16/libs/challenge-registry/ui/src/lib/challenge-card/mock-challenges.ts#L19) for the 4 mock challenges.

Yes, it's expected with this simple mock-up service, which tries to search for the input string in any properties of challenge. I think @tschaffter has merged the real better service to filter challenges by search term today. We will replace with real service later.

> Are we still using the fullName attribute? I don't see it listed anymore in [challenge.openapi.yaml](https://github.com/Sage-Bionetworks/sage-monorepo/blob/main/libs/challenge-registry/api-description/build/challenge.openapi.yaml).

`fullname` was used to define the url path (like {org_name}/{challenge_name}) of the challenge page. I don't remember why we needed it in the legacy repo, but we probably don't not need the `fullname` property anymore. Maybe @tschaffter could answer this question.

Reply:
> We will replace with real service later.

That's right, the challenge service now supports full-text search (#1104).

> Are we still using the fullName attribute? I don't see it listed anymore in [challenge.openapi.yaml](https://github.com/Sage-Bionetworks/sage-monorepo/blob/main/libs/challenge-registry/api-description/build/challenge.openapi.yaml).

No, `fullName` is no longer used.